File: algorithms/237.Delete_Node_in_a_Linked_List/Delete_Node_in_a_Linked_List.cpp
<<<<<<< HEAD
#include <bits/stdc++.h>

using namespace std;

// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};


class Solution {
public:
    void deleteNode(ListNode* node) {
        ListNode *next = node->next;
        *node = *next;
        // this statement is equivalent to
        // node->val = next->val;
        // node->next = next->next;
        delete next;
    }
};
class Solution {
public:
   void deleteNode(ListNode* node) {
    auto next = node->next;
    *node = *next;
    delete next;
}
};

int main() {

    return 0;
}

=======
#include <bits/stdc++.h>

using namespace std;

// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};


class Solution {
public:
    void deleteNode(ListNode* node) {
        ListNode *next = node->next;
        *node = *next;
        // this statement is equivalent to
        // node->val = next->val;
        // node->next = next->next;
        delete next;
    }
};
class Solution {
public:
   void deleteNode(ListNode* node) {
    auto next = node->next;
    *node = *next;
    delete next;
}
};

int main() {

    return 0;
}

>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74

File: algorithms/237.Delete_Node_in_a_Linked_List/Delete_Node_in_a_Linked_List.java
<<<<<<< HEAD
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public void deleteNode(ListNode node) {
    node.val = node.next.val;
    node.next = node.next.next;
}
}
=======
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public void deleteNode(ListNode node) {
    node.val = node.next.val;
    node.next = node.next.next;
}
}
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
